inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,228 @@ pheatmap(
     annotation_col = as.data.frame(sample_metadata))
 ```
 
+### *bonus* Regression charts to visualize microbiome data
+
+Regression analysis is a powerful statistical method that can help examine
+the relationship between two or more variables. In microbiome research,
+regression charts help in visualizing how changes in microbial communities
+correlate with various environmental factors, health outcomes, or other
+biological variables.
+
+Let us do a regression analysis exercise using the `GlobalPatterns` dataset
+against the Shannon index ("ShannonIndex"), a common measure of alpha diversity,
+which quantifies the diversity within each sample. This index will serve as the
+dependent variable in our regression model examples.
+
+We extract "SampleType" as a categorical(discrete) variable and "coverage" as a
+continuous variable from the dataset’s metadata. These variables will be used
+as predictors/independent variables in our linear regression models.
+
+1. Load the necessary libraries and import the dataset into R; here we are
+using a microbiome dataset.
+
+```{r}
+#| label: extra-regressionchart-step1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+library(mia)             
+library(TreeSummarizedExperiment) 
+library(scater) 
+library(lmtest)
+library(gtsummary)   
+library(jtools)   
+      
+data("GlobalPatterns")
+tse <- GlobalPatterns
+```
+
+2. Calculate Alpha Diversity (Shannon Index), i.e. the dependent variable.
+
+```{r}
+#| label: extra-regressionchart-step2
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+tse <- addAlpha(tse)
+shannon <- colData(tse)$shannon
+```
+
+Now let's define the covariates i.e. the independent variables.
+
+3. First, fit linear model using a discrete variable, say "SampleType", where 
+Shannon diversity is the response variable, and SampleType is the predictor. 
+
+"SampleType" represents the type or source of the microbiome sample. It 
+categorizes each sample based on the environment or biological material from
+which it was collected.
+
+```{r}
+#| label: extra-regressionchart-step3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
```

Comment:
"eval: true" should be default - is there a reason to not evaluate this? check the same for other chunks